src/iceberg/avro/CMakeLists.txt

```
@@ -28,20 +28,39 @@ set(ICEBERG_AVRO_SHARED_BUILD_INTERFACE_LIBS)
 set(ICEBERG_AVRO_STATIC_INSTALL_INTERFACE_LIBS)
 set(ICEBERG_AVRO_SHARED_INSTALL_INTERFACE_LIBS)
 
-list(APPEND ICEBERG_AVRO_STATIC_BUILD_INTERFACE_LIBS
+list(APPEND
+     ICEBERG_AVRO_STATIC_BUILD_INTERFACE_LIBS
      "$<IF:$<TARGET_EXISTS:iceberg_static>,iceberg_static,iceberg_shared>"
-     Avro::avro_static)
-list(APPEND ICEBERG_AVRO_SHARED_BUILD_INTERFACE_LIBS
+     "$<IF:$<TARGET_EXISTS:Avro::avrocpp_static>,Avro::avrocpp_static,Avro::avrocpp_shared>"
+)
+list(APPEND
+     ICEBERG_AVRO_SHARED_BUILD_INTERFACE_LIBS
      "$<IF:$<TARGET_EXISTS:iceberg_shared>,iceberg_shared,iceberg_static>"
-     Avro::avro_shared)
+     "$<IF:$<TARGET_EXISTS:Avro::avrocpp_shared>,Avro::avrocpp_shared,Avro::avrocpp_static>"
+)
+
+if(ARROW_VENDORED)
```

I guess here should be AVRO_VENDORED?
```suggestion
if(AVRO_VENDORED)
```

---

Good catch!